### PR TITLE
refactor(SPV-1174):  add user contacts, invitation api implementations

### PR DESCRIPTION
**Pull request by mgosek-4chain**
# Pull Request Checklist

- [X] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_STANDARDS.md)
- [X] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-go-client/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] 🏠 I tested my changes locally.
- [X] ✅ I have provided tests for my changes.
- [X] 📝 I have used conventional commits.
- [X] 📗 I have updated any related documentation.
- [X] 💾 PR was issued based on the Github or Jira issue.

> [!IMPORTANT] 
> Before running the implementation with a manual example, ensure that the SPV Wallet API is running locally in your environment.

**Options to check (user contacts API):**
-  ContactQueryWithMetadataFilter
-  ContactQueryWithPageFilter
-  ContactQueryWithContactFilter

**Example testing code snippet:**
```
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"strings"

	walletclient "github.com/bitcoin-sv/spv-wallet-go-client"
	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
	"github.com/bitcoin-sv/spv-wallet/models/filter"
)

func main() {
	spv, err := walletclient.NewWithXPriv(ExampleConfig, ExampleXPriv)
	if err != nil {
		log.Fatal(err)
	}

	status := "unconfirmed"
	contacts, err := spv.Contacts(context.Background(), queries.ContactQueryWithContactFilter(filter.ContactFilter{
		Status: &status,
	}))
	if err != nil {
		log.Fatal(err)
	}

	Print("Contacts:", contacts)
}

var ExampleConfig = walletclient.NewDefaultConfig("http://localhost:3003")

const (
	ExampleXPriv string = "xprv"
	ExampleXPub  string = ""
)

func Print(s string, a any) {
	fmt.Println(strings.Repeat("~", 100))
	fmt.Println(s)
	fmt.Println(strings.Repeat("~", 100))
	res, err := json.MarshalIndent(a, "", " ")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(res))
}
```

**User invitations API methods to check:**
-  AcceptInvitation
-  RejectInvitation
 
**Example testing code snippet:**
```
package main

import (
	"context"
	"log"

	walletclient "github.com/bitcoin-sv/spv-wallet-go-client"
)

func main() {
	spv, err := walletclient.NewWithXPriv(ExampleConfig, ExampleXPriv)
	if err != nil {
		log.Fatal(err)
	}

	paymail := "john.doe@test.com"
	err = spv.AcceptInvitation(context.Background(), paymail)
	if err != nil {
		log.Fatal(err)
	}
}

var ExampleConfig = walletclient.NewDefaultConfig("http://localhost:3003")

const (
	ExampleXPriv string = "xprv"
	ExampleXPub  string = ""
)
```

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -129,7 +233,9 @@ type authenticator interface {
 func newClient(cfg Config, auth authenticator) *Client {
 	restyCli := newRestyClient(cfg, auth)
```
Comment (dzolt-4chain) [inline client.go]:
```suggestion
	restyClient := newRestyClient(cfg, auth)
```

File: client.go [hunk 2]
```diff
@@ -129,7 +233,9 @@ type authenticator interface {
...
 	cli := Client{
```
Comment (dzolt-4chain) [inline client.go]:
```suggestion
	client := Client{
```

File: client.go [hunk 3]
```diff
@@ -129,7 +233,9 @@ type authenticator interface {
...
-		configsAPI: configs.NewAPI(cfg.Addr, restyCli),
+		configsAPI:     configs.NewAPI(cfg.Addr, restyCli),
+		contactsAPI:    contacts.NewAPI(cfg.Addr, restyCli),
+		invitationsAPI: invitations.NewAPI(cfg.Addr, restyCli),
 	}
 	return &cli
```
Comment (dzolt-4chain) [inline client.go]:
```suggestion
		configsAPI:     configs.NewAPI(cfg.Addr, restyClient),
		contactsAPI:    contacts.NewAPI(cfg.Addr, restyClient),
		invitationsAPI: invitations.NewAPI(cfg.Addr, restyClient),
	}
	return &client
```

File: internal/api/v1/user/contacts/contacts.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import (
+	"context"
+	"fmt"
+
+	"github.com/bitcoin-sv/spv-wallet-go-client/commands"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/go-resty/resty/v2"
+)
+
+// TODO: 1. Contacts should accept the optional query parameters similar to the transactions.
```
Comment (dzolt-4chain) [inline internal/api/v1/user/contacts/contacts.go]: don't forget to rm todo comment, unless you explicitly want it to stay here?

File: internal/api/v1/user/contacts/contacts.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+		R().
+		SetBody(r).
+		SetContext(ctx).
+		SetResult(&result).
+		Put(URL)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP response failure: %w", err)
+	}
+
+	return &response.Contact{
```
Comment (dzolt-4chain) [inline internal/api/v1/user/contacts/contacts.go]: What's the purpose of this mapping? Can't we return just result.Contact here?

File: internal/api/v1/user/contacts/contacts_api.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	queryBuilder := querybuilders.NewQueryBuilder([]querybuilders.QueryBuilderOption{
+		querybuilders.WithMetadataFilter(query.Metadata),
+		querybuilders.WithPageFilter(query.PageFilter),
+		querybuilders.WithFilterQueryBuilder(&contactFilterQueryBuilder{
+			contactFilter: query.ContactFilter,
+			modelFilterBuilder: querybuilders.ModelFilterBuilder{
+				ModelFilter: query.ContactFilter.ModelFilter,
+			},
+		}),
+	}...)
```
Comment (chris-4chain) [inline internal/api/v1/user/contacts/contacts_api.go]: Why dont to put these options right into args?

```suggestion
	queryBuilder := querybuilders.NewQueryBuilder(
		querybuilders.WithMetadataFilter(query.Metadata),
		querybuilders.WithPageFilter(query.PageFilter),
		querybuilders.WithFilterQueryBuilder(&contactFilterQueryBuilder{
			contactFilter: query.ContactFilter,
			modelFilterBuilder: querybuilders.ModelFilterBuilder{
				ModelFilter: query.ContactFilter.ModelFilter,
			},
		}),
	)
```

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**